api/spec/src/subscription.tsp

```diff
@@ -0,0 +1,390 @@
+     */
+    @query at?: DateTime,
+  ): Subscription | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  @get
+  @operationId("getSubscription")
+  @sharedRoute
+  getExpanded(
+    @path subscriptionId: ULID,
+    @query expand: true,
+
+    /**
+     * The time at which the subscription should be queried. If not provided the current time is used.
+     */
+    @query at?: DateTime,
+  ): SubscriptionExpanded | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  create(
+    @body body: {
+      customerId: ULID;
+      templatingPlan: {
+        planKey: Key;
+        planVersion: integer;
+      };
+      currency: CurrencyCode;
+      activeFrom: DateTime;
+
+      /**
+       * JSON Patch (RFC6902).
+       *
+       * It is a partial implementation of the JSON Patch standard:
+       * 1. It only respecting the `add` and `remove` operations. All other operations defined in RFC6902 are ignored during execution.
+       * 2. The provided value for ADD operation doesn't match the target type. It is the creation input type for the SubscriptionItem.
+       */
+      @maxItems(100)
+      overrides: SubscriptionItemOverride[];
+    },
+  ): Subscription | OpenMeter.ConflictError | OpenMeter.CommonErrors;
+
+  /**
+   * Applies the JSON Patch (RFC6902) atomically (RFC5789).
+   *
+   * It is a partial implementation of the JSON Patch standard:
+   * 1. It only respecting the `add` and `remove` operations. All other operations defined in RFC6902 are ignored during execution.
+   * 2. The provided value for ADD operation doesn't match the target type. It is the creation input type for the SubscriptionItem and SubscriptionPhase.
+   */
+  @patch
+  @operationId("editSubscription")
+  @route("/{subscriptionId}")
+  edit(
+    @header accept: "json-patch+json",
+    @path subscriptionId: ULID,
+
+    @body
+    @maxItems(100)
+    body: SubscriptionItemOverride[],
+  ): Subscription | OpenMeter.ConflictError | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  /**
+   * Migrates the subscripiton to the procided version of the plan.
+   */
+  @post
+  @operationId("migrateSubscription")
+  @route("/{subscriptionId}/migrate")
+  migrate(
+    @path subscriptionId: ULID,
+    @body body: {
+      @minValue(1)
+      targetVersion: integer;
+    },
+  ): Subscription | OpenMeter.ConflictError | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  /**
+   * Cancels the subscription.
+   */
+  @post
+  @operationId("cancelSubscription")
+  @route("/{subscriptionId}/cancel")
+  cancel(
+    @path subscriptionId: ULID,
+    @body body: {
+      /**
+       * If not provided the subscription is canceled immediately.
+       */
+      effectiveDate?: DateTime;
+    },
+  ): Subscription | OpenMeter.ConflictError | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  /**
+   * Cancels the scheduled cancelation.
+   */
+  @post
+  @operationId("cancelScheduledCancelation")
+  @route("/{subscriptionId}/cancel-scheduled-cancelation")
```

Comment:
mounthful

Reply:
we could do `@route("/{subscriptionId}/unschedule-cancelation")`